src/Data/Foldable.purs

```diff
@@ -325,6 +327,30 @@ elem = any <<< (==)
 notElem :: forall a f. Foldable f => Eq a => a -> f a -> Boolean
 notElem x = not <<< elem x
 
+-- | Try to get nth element from the left in a data structure
+indexl :: forall a f. Foldable f => Int -> f a -> Maybe a
+indexl idx = _.elem <<< foldl go { elem: Nothing, pos: 0 }
+  where
+  go cursor elem =
+    case cursor.elem of 
+      Just _ -> cursor
+      _ ->
+        if cursor.pos == idx
+          then { elem: Just elem, pos: cursor.pos }
+          else { pos: cursor.pos + 1, elem: cursor.elem }
+
+-- | Try to get nth element from the right in a data structure
+indexr :: forall a f. Foldable f => Int -> f a -> Maybe a
+indexr idx = _.elem <<< foldr go { elem: Nothing, pos: 0 }
+  where
+  go elem cursor =
```

Comment:
I could create non-exported indexGo function and use it in in both places (in one I would need to flip it)